presto-parser/src/main/java/com/facebook/presto/sql/tree/DefaultTraversalVisitor.java

```diff
@@ -484,13 +482,30 @@ protected R visitGroupBy(GroupBy node, C context)
     }
 
     @Override
-    protected R visitGroupingElement(GroupingElement node, C context)
+    protected R visitCube(Cube node, C context)
+    {
+        return null;
+    }
+
+    @Override
+    protected R visitRollup(Rollup node, C context)
+    {
+        return null;
+    }
+
+    @Override
+    protected R visitSimpleGroupBy(SimpleGroupBy node, C context)
     {
-        for (Set<Expression> expressions : node.enumerateGroupingSets()) {
-            for (Expression expression : expressions) {
-                process(expression, context);
-            }
+        for (Expression expression : node.getColumnExpressions()) {
```

Comment:
Why simple group by is so special? Maybe a comment would be nice.

Reply:
SimpleGroupBy allows arbitrary expressions, so the visitor needs to process them recursively.